File: solutions/0242-valid-anagram/solution.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.length() != t.length()) {
            return false;
        }
        std::vector<short> v(26, 0);
        for (auto c: s) {
            v[c - 'a']++;
        }
        for (auto c: t) {
            v[c - 'a']--;
        }
        for (auto i: v) {
            if (i > 0) {
                return false;
            }
        }
        return true;
    }
};

File: solutions/0242-valid-anagram/solution.go
func isAnagram(s string, t string) bool {
    if len(s) != len(t) {
        return false
    }
    m := make(map[rune]int)
    for _, c := range s {
        v, exists := m[c]
        if exists {
            m[c] = v + 1
        } else {
            m[c] = 1
        }
    }
    for _, c := range t {
        v, exists := m[c]
        if !exists {
            return false
        }
        m[c] = v - 1
        if m[c] == 0 {
            delete(m, c)
        }
    }
    return len(m) == 0
}
